### ISSUE TITLE
AttributeError: 'dict' object has no attribute 'split'

### ISSUE THREAD

**Issue by natsukmoe** [mkdocs/mkdocs#2156]
When I tried to build my pages, an error occured. The full output below:
```
INFO    -  Building documentation...
INFO    -  Cleaning site directory
Traceback (most recent call last):
  File "/usr/local/bin/mkdocs", line 11, in <module>
    sys.exit(cli())
  File "/usr/lib/python3/dist-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3/dist-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3/dist-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3/dist-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/mkdocs/__main__.py", line 136, in serve_command
    **kwargs
  File "/usr/local/lib/python3.6/dist-packages/mkdocs/commands/serve.py", line 141, in serve
    config = builder()
  File "/usr/local/lib/python3.6/dist-packages/mkdocs/commands/serve.py", line 136, in builder
    build(config, live_server=live_server, dirty=dirty)
  File "/usr/local/lib/python3.6/dist-packages/mkdocs/commands/build.py", line 285, in build
    _build_theme_template(template, env, files, config, nav)
  File "/usr/local/lib/python3.6/dist-packages/mkdocs/commands/build.py", line 108, in _build_theme_template
    output = _build_template(template_name, template, files, config, nav)
  File "/usr/local/lib/python3.6/dist-packages/mkdocs/commands/build.py", line 87, in _build_template
    output = template.render(context)
  File "/usr/local/lib/python3.6/dist-packages/jinja2/environment.py", line 1090, in render
    self.environment.handle_exception()
  File "/usr/local/lib/python3.6/dist-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/usr/local/lib/python3.6/dist-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/usr/local/lib/python3.6/dist-packages/material/404.html", line 4, in top-level template code
    {% extends "base.html" %}
  File "/usr/local/lib/python3.6/dist-packages/material/base.html", line 112, in top-level template code
    {% block header %}
  File "/usr/local/lib/python3.6/dist-packages/material/base.html", line 113, in block "header"
    {% include "partials/header.html" %}
  File "/usr/local/lib/python3.6/dist-packages/material/partials/header.html", line 7, in top-level template code
    {% include "partials/logo.html" %}
  File "/usr/local/lib/python3.6/dist-packages/material/partials/logo.html", line 5, in top-level template code
    <img src="{{ config.theme.logo | url }}" alt="logo">
  File "/usr/local/lib/python3.6/dist-packages/mkdocs/utils/filters.py", line 14, in url_filter
    return normalize_url(value, page=context['page'], base=context['base_url'])
  File "/usr/local/lib/python3.6/dist-packages/mkdocs/utils/__init__.py", line 262, in normalize_url
    path = path_to_url(path or '.')
  File "/usr/local/lib/python3.6/dist-packages/mkdocs/utils/__init__.py", line 290, in path_to_url
    return '/'.join(path.split('\\'))
AttributeError: 'dict' object has no attribute 'split'
```

**Comment by facelessuser** [comment]
In order for anyone to help you, you would need to provide, at the very least, your config as most likely it is misconfigured causing this issue.

**Comment by natsukmoe** [comment]
OK. I've tried 3 times in 3 different environments. The versions of all mkdocs are 1.1.2, the newest.
1. Ubuntu 18.04 Server
2. Ubuntu 20.04 Desktop
3. macOS 10.15 Catalina

My configure file mkdocs.yml below:
(I've installed the material theme)
```
# Project Information
site_name: Name
site_description: Description
site_author: Author
# site_url: Url

# Contents
nav:
  - About:
    - Welcome: index.md

# Theme
theme:
  name: 'material'
  language: 'zh'
  palette:
    primary: 'white'
    accent: 'red'
  include_search_page: false
  search_index_only: true
  favicon: 'icon.ico'

  logo:
    icon: 'gamepad'
  feature:
    tabs: true
  font:
    text: 'Fira Sans'
    code: 'Fira Mono'
```

The directions are:
```
project
|-docs
||-index.md
|-mkdocs.yml
|-icon.ico
```

**Comment by natsukmoe** [comment]
I've tried to build another project.

It is a completed project and has been uploaded to a website.

But I also met this error.

**Comment by jimporter** [comment]
This looks like it's really an issue with the Material theme, not MkDocs in general, but I think this is the problem:

```
  logo:
    icon: 'gamepad'
```

The [Material docs](https://squidfunk.github.io/mkdocs-material/getting-started/#logo) say that `logo` should be a string pointing to the path of an image, e.g. `logo: images/my_logo.png`.

**Comment by natsukmoe** [comment]
I've changed it to
```
theme:
  name: 'material'
  language: 'zh'
  palette:
    primary: 'white'
    accent: 'red'
  include_search_page: false
  search_index_only: true
  favicon: 'icon.ico'

  logo:
    icon: 'icon.png'
  feature:
    tabs: true
  font:
    text: 'Fira Sans'
    code: 'Fira Mono'
```
but the error still occurs.

**Comment by NicholasYamamoto** [comment]
@natsukmoe 

Hello! I was having the same issue when trying to run `mkdocs serve`, and discovered that there were [breaking changes made between versions 4.x and 5.x of the 'mkdocs-material' dependency](https://github.com/squidfunk/mkdocs-material/issues/1576#issuecomment-613483837).

I fixed this by following the 4.x to 5.x upgrade guide on the official `mkdocs` Github repo [here](https://github.com/squidfunk/mkdocs-material/issues/1576#issuecomment-613483837).


In short, run the following command to upgrade your version of `mkdocs-material` to the latest:
```zsh
pip install --upgrade mkdocs-material`
```

Then you can run the following command to verify the current version of `mkdocs-material`:
```zsh
pip show mkdocs-material`
```

This is what it took for me to fix this issue, so hopefully it works for you too!

**Comment by waylan** [comment]
Closing this as i appears to be related to the material theme. For material theme issues, please go to that project.

**Comment by ARanz-lolo** [comment]
<script>alert(1)</script>

**Comment by ARanz-lolo** [comment]
/>XSS Test
